docs/proposals/operator-sdk-v1-migration.md

```diff
@@ -0,0 +1,285 @@
+<!--
+Copyright The Shipwright Contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+---
+title: crd-domain-rename
+authors:
+  - @adambkaplan
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-12-08
+last-updated: 2020-12-09
+status: provisional
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Operator SDK v1 Migration
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Open Questions
+
+1. Do we declare the migration to SDK v1 a breaking update? Do we not support upgrades from v0.2.x
+   to v0.3.0?
+2. Should we move our base domain to `shipwright.io`?
```

Comment:
Let's go for this.


Can we migrate existing CRs this way:
1. Add a controller that recreates CRs with the new group
2. Remove it after **one** upstream release.
3. CodeEngine needs to ensure they don't jump versions while upgrading? :)

Reply:
IIRC Custom resource definitions are not removed by OLM, even if the operator is removed.

If we use a different ConfigMap for leader election locking, then the old and new controllers can live side-by-side because the APIs have different GroupVersions.

Reply:
> I´m assuming this is about the group.

Yes, ultimately it trickles down to the full GroupVersionKind for our APIs.

Changing this to `shipwright.io` is important for branding and the general experience of using Shipwright builds. Using our shipwright.io domain makes it clear to administrators that they are running Shipwright components on their clusters, and likewise it helps developers identify they are using Shipwright when they create Build YAML manifests. Seeing "shipwright" everywhere establishes and reinforces the brand.

`build.dev`, on the other hand, is a currently open domain that we have no control over. Google owns the `.dev` TLD, and hence why Tekton is able to use it. Conceivably someone could purchase it and provide their own "build API" for Kubernetes, or something else entirely. Google could also decide to purchase it back and use it for their own purposes.

In any case, this is something we do not _have_ to do as part of the operator-sdk upgrade. We can take this on in a subsequent release that we declare as a "breaking" release.

Reply:
ok, I will open an issue for this topic. I would prefer to not do this breaking change yet, but rather discuss on a community meeting and consolidate the plan in the upcoming issue, resolving the conversation.